### PR TITLE
Make admins onsite users by default for all programs

### LINKED ISSUE #2835: Onsite programs list not loading
Northeastern reported that https://neptun.learningu.org/myesp/onsite/ failed to load the upcoming program. It said there were multiple upcoming programs but then didn't link to any, as it should have. Controlling code is here https://github.com/learning-unlimited/ESP-Website/blob/main/esp/esp/web/views/myesp.py#L261. Unclear if the "TODO" comment if related.

### PR DESCRIPTION
**Pull request by milescalabresi**
Fixes #2835

### CODE REVIEW

**Comment by willgearty** [comment]
I thought we were going to give admins the "Admin" permission? That should include the "Onsite" permission, but also all of the other permissions.

File: esp/esp/program/setup.py [hunk 1]
```diff
@@ -76,24 +73,23 @@ def gen_perm(tup):
...
-        #It's not for a specific user and not a teacher or student deadline
-        for x in ESPUser.getTypes():
-            newnew_perm=Permission(permission_type=new_perm.permission_type, role=Group.objects.get(name=x), start_date=new_perm.start_date, end_date=new_perm.end_date, program=prog)
-            newnew_perm.save()
 
     for perm_tup in perms:
         gen_perm(perm_tup)
+    
+    # Grant onsite permission to admins and onsite user (if an onsite user is available).
+    Permission(permission_type='Onsite', role=Group.objects.get(name="Administrator"), start_date=None, end_date=None, program=prog).save()
```
Comment (willgearty) [inline esp/esp/program/setup.py]: I thought we were going to give admins the "Admin" permission? That should include the "Onsite" permission, but also all of the other permissions.

File: esp/esp/web/views/myesp.py [hunk 2]
```diff
@@ -264,6 +264,9 @@ def myesp_onsite(request):
     # - Currently reverse the list in Python, otherwise fbap's cache is ignored
     # TODO: Fix this
     progs = list(progs)
+    if len(progs) > 0:
+        # filter out past programs; admins can always go to those through /manage/programs to access their onsite pages
+        progs = [p for p in progs if p.datetime_range() is None or len(p.datetime_range()) == 0 or p.datetime_range()[-1] > datetime.datetime.now()]
```
Comment (willgearty) [inline esp/esp/web/views/myesp.py]: Do we really want to include programs without dates?

File: esp/esp/program/setup.py (hunk 1 above)
Comment (milescalabresi) [reply esp/esp/program/setup.py]: Does it? I thought part of the problem was it didn't, but I will double-check

File: esp/esp/web/views/myesp.py (hunk 2 above)
Comment (milescalabresi) [reply esp/esp/web/views/myesp.py]: Yeah, I thought about it, and for test programs or programs where people manage to goof something up, I think we still want admins to have full access

File: esp/esp/program/setup.py (hunk 1 above)
Comment (willgearty) [reply esp/esp/program/setup.py]: I'm pretty sure the "Administer" permission gives *all* permissions for a program:
https://github.com/learning-unlimited/ESP-Website/blob/a0a1e50fbee39598313869d917f43ffd889cc30b/esp/esp/users/models/__init__.py#L2326

File: esp/esp/web/views/myesp.py (hunk 2 above)
Comment (willgearty) [reply esp/esp/web/views/myesp.py]: But if a program doesn't have dates, will any of the onsite stuff even make sense/work?

**Comment by willgearty** [comment]
This seems relevant: https://github.com/learning-unlimited/ESP-Website/issues/2374

**Pull request closed without merging**